service/history/replicatorQueueProcessor.go

```diff
@@ -304,6 +304,13 @@ func (p *replicatorQueueProcessorImpl) getTasks(
 
 	if lastReadTaskID == persistence.EmptyQueueMessageID {
 		lastReadTaskID = p.shard.GetClusterReplicationLevel(pollingCluster)
+	} else {
+		if err := p.shard.UpdateClusterReplicationLevel(
```

Comment:
Do we really want to `UpdateClusterReplicationLevel` on every read?

Reply:
Looking at the logic looks like we were doing it for every replication task which was even worse.  Let's have a quick chat tomorrow to see if we can move it to a back ground task and do it even less frequently then how it is coded up right now.

Reply:
https://github.com/temporalio/temporal/blob/622852f5c1abaae5e1f966fc532d5f74660c170c/service/history/shard/context_impl.go#L895

there is check preventing too many updates to DB
in mem change is fine

Reply:
this getTasks will only be called when target cluster calling source cluster

Reply:
> Looking at the logic looks like we were doing it for every replication task which was even worse.
not each replication task, but each get tasks call